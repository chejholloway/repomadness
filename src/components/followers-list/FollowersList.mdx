# FollowersList Component

Renders a list of followers with their avatars.

This component takes an array of `Follower` objects as a prop and renders
a list of avatars with links to their profiles.

## Component Details

- **Type:** React Functional Component
- **Props:**

  - `followers`: An array of `Follower` objects representing followers.

## Usage Example

```tsx
import React from 'react';
import { Link } from 'react-router-dom';
import { Avatar, Typography } from '@material-tailwind/react';

import Follower from '../../types/Follower';

const FollowersList: React.FC<FollowersListProps> = ({ followers }) => {
  return (
    <div className='text-gray-700'>
      <h3 className='text-center'>
        <Typography className='font-normal'>Followers</Typography>
      </h3>
      <div className='flex flex-wrap space-x-4'>
        {followers ? (
          followers.map((follower) => (
            <div key={follower.id}>
              <Link to={'/'}>
                <Avatar
                  variant='circular'
                  title={`${follower?.login}'s Avatar`}
                  alt={follower?.login}
                  src={follower?.avatar_url}
                />
              </Link>
            </div>
          ))
        ) : (
          <p>No followers available.</p>
        )}
      </div>
    </div>
  );
};

export default FollowersList;
