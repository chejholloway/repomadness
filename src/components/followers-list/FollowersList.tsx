import React from "react";
import Follower from "../../types/Follower";
import FollowerCard from "../cards/FollowerCard";
import GenericList from "../generic-list/GenericList";

interface FollowersListProps {
  followers: Follower[] | undefined;
}

const FollowersList: React.FC<FollowersListProps> = ({ followers }) => {
  const mapToLink = (follower: Follower) => follower.html_url;
  const renderItem = (follower: Follower) => (
    <FollowerCard follower={follower} />
  );
  const renderCard = (follower: Follower) => (
    <FollowerCard follower={follower} />
  );

  return (
    <GenericList
      title="Followers"
      data={followers}
      renderItem={renderItem}
      emptyMessage="No followers available."
      mapToLink={mapToLink}
      renderCard={renderCard}
    />
  );
};

export default FollowersList;
