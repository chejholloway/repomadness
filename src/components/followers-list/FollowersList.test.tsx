import React from "react";
import { render } from "@testing-library/react";

import FollowersList from "./FollowersList";

// Mock react-router-dom and @material-tailwind/react components
jest.mock("react-router-dom", () => ({
  Link: ({ children }: { children: React.ReactNode }) => children,
}));

jest.mock("@material-tailwind/react", () => ({
  Avatar: () => null,
  Typography: () => null,
}));

// Mock the Follower type
jest.mock("../../types/Follower");

const mockFollowers = [
  {
    login: "mountain",
    id: 851,
    node_id: "MDQ6VXNlcjg1MQ==",
    avatar_url: "https://avatars.githubusercontent.com/u/851?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/mountain",
    html_url: "{process.env.REACT_APP_GITHUB_URL}/mountain",
    followers_url: "https://api.github.com/users/mountain/followers",
    following_url:
      "https://api.github.com/users/mountain/following{/other_user}",
    gists_url: "https://api.github.com/users/mountain/gists{/gist_id}",
    starred_url: "https://api.github.com/users/mountain/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/mountain/subscriptions",
    organizations_url: "https://api.github.com/users/mountain/orgs",
    repos_url: "https://api.github.com/users/mountain/repos",
    events_url: "https://api.github.com/users/mountain/events{/privacy}",
    received_events_url:
      "https://api.github.com/users/mountain/received_events",
    type: "User",
    site_admin: false,
  },
  {
    login: "sunfmin",
    id: 1014,
    node_id: "MDQ6VXNlcjEwMTQ=",
    avatar_url: "https://avatars.githubusercontent.com/u/1014?v=4",
    gravatar_id: "",
    url: "https://api.github.com/users/sunfmin",
    html_url: "{process.env.REACT_APP_GITHUB_URL}/sunfmin",
    followers_url: "https://api.github.com/users/sunfmin/followers",
    following_url:
      "https://api.github.com/users/sunfmin/following{/other_user}",
    gists_url: "https://api.github.com/users/sunfmin/gists{/gist_id}",
    starred_url: "https://api.github.com/users/sunfmin/starred{/owner}{/repo}",
    subscriptions_url: "https://api.github.com/users/sunfmin/subscriptions",
    organizations_url: "https://api.github.com/users/sunfmin/orgs",
    repos_url: "https://api.github.com/users/sunfmin/repos",
    events_url: "https://api.github.com/users/sunfmin/events{/privacy}",
    received_events_url: "https://api.github.com/users/sunfmin/received_events",
    type: "User",
    site_admin: false,
  },
];

describe("<FollowersList>", () => {
  it("should render component", () => {
    const { asFragment } = render(<FollowersList followers={mockFollowers} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render component with props", () => {
    const { asFragment } = render(<FollowersList followers={mockFollowers} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
