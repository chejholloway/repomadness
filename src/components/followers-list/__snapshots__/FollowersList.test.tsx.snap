// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FollowersList> should render component 1`] = `
<DocumentFragment>
  <div
    class="css-818d2q"
  >
    <h1
      class="font-normal css-qhut37"
    >
      Followers
    </h1>
    <div
      class="flex flex-wrap space-x-4"
    >
      <div
        class="m-4"
      />
      <div
        class="m-4"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<FollowersList> should render component with props 1`] = `
<DocumentFragment>
  <div
    class="css-818d2q"
  >
    <h1
      class="font-normal css-qhut37"
    >
      Followers
    </h1>
    <div
      class="flex flex-wrap space-x-4"
    >
      <div
        class="m-4"
      />
      <div
        class="m-4"
      />
    </div>
  </div>
</DocumentFragment>
`;
